store/tikv/coprocessor.go

```diff
@@ -677,6 +683,47 @@ func (worker *copIteratorWorker) handleTaskOnce(bo *Backoffer, task *copTask, ch
 	return worker.handleCopResponse(bo, rpcCtx, &copResponse{pbResp: resp.Resp.(*coprocessor.Response)}, task, ch, nil, costTime)
 }
 
+// clientHelper wraps LockResolver and RegionRequestSender.
+// It's introduced to support the new lock resolving pattern in the large transaction.
+// In the large transaction protocol, sending requests and resolving locks are
+// context-dependent. For example, when a send request meets a secondary lock, we'll
+// call ResolveLock, and if the lock belongs to a large transaction, we may retry
+// the request. If there is no context information about the resolved locks, we'll
+// meet the secondary lock again and run into a deadloop.
+type clientHelper struct {
+	*LockResolver
+	*RegionRequestSender
+
+	minCommitTSPushed map[RegionVerID][]uint64
```

Comment:
I think we don't need to separate the transactions by region.
A transaction may cross multiple regions.
Put all resolved transaction ids will reduce the chance that a lock already resolved being returned.

Reply:
ok